email_totals/ses.py

```diff
@@ -106,6 +322,11 @@ def send_report_email(recipient, body_html, body_text):
             Source=sender,
         )
 
+        # We need to rate limit our calls to 'send_email', our current SES
+        # quota allows for sending 14 emails per second, sleep for 72ms after
+        # each call to send a maximum of 14 emails in 1008ms.
+        time.sleep(0.072)
```

Comment:
was this the recommended approach to rate limit?  if so could you please provide a comment with link to the source?

Reply:
AWS doesn't offer any recommendations that I can find on implementing client-side rate-limiting, but while looking again I've found a user guide describing [boto's backoff/retry configuration](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html). So I'll add a retry config block to improve handling rate limit exceptions from the API in case my naive implementation here still runs too frequently